### PR TITLE
SQLAlchemy v2 support

### PR DESCRIPTION
**Pull request by smotornyuk**
Compatibility with the new SQLAlchemy version tested against [this PR](https://github.com/ckan/ckan/pull/8410).

### CODE REVIEW

**Comment by duttonw** [comment]
Looks like setup tools for 2.9 has started to fail unrelated to your changes. 


      /usr/lib/python3.9/site-packages/setuptools/_distutils/dist.py:261: UserWarning: Unknown distribution option: 'tests_require'

**Comment by ThrawnCA** [comment]
> Looks like setup tools for 2.9 has started to fail unrelated to your changes.

We can fix that, but I'll note that with the release of CKAN 2.11, 2.9 is no longer officially supported: https://ckan.org/blog/the-new-ckan-211-release-is-now-available

**Comment by ThrawnCA** [comment]
Looks like 2.9 isn't compatible with the newer SQLAlchemy. We should probably make version-specific requirements files.

**Comment by duttonw** [comment]
I've cut a release today for 1.2.0 which is compatible for 2.9 ~ 2.11. 

I'm ok with dropping 2.9 support via this PR, I've updated the doc's and cicd pipeline accordingly. 

I'd like to pull in the last pieces of work from the qld-gov-au fork prior to this being included for one last 1.x.x release.

As this is major breaking change, it should bump to 2.0.0 on this being merged into master/released.

**Review by ThrawnCA — APPROVED** [review]

**Review by ThrawnCA — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -53,8 +53,6 @@ jobs:
             ckan-image: "2.11-py3.10"
           - ckan-version: "2.10"
             ckan-image: "2.10-py3.10"
-          - ckan-version: "2.9"
```
Comment (ThrawnCA) [inline .github/workflows/publish.yml:56]: Still needs to be removed from the test workflow.

**Review by ThrawnCA — CHANGES_REQUESTED** [review]

**Review by ThrawnCA — APPROVED** [review]

**Pull request merged**